tests/opentracer/test_tracer.py

```diff
@@ -308,22 +308,23 @@ def test_start_span_multi_intertwined(self, ot_tracer, writer):
         event = threading.Event()
 
         def trace_one():
-            id = 11  # noqa: A001
-            with ot_tracer.start_active_span(str(id)):
-                id += 1
-                with ot_tracer.start_active_span(str(id)):
-                    id += 1
-                    with ot_tracer.start_active_span(str(id)):
-                        event.set()
+            _id = 11
+            with ot_tracer.start_active_span(str(_id)):
+                _id += 1
+                with ot_tracer.start_active_span(str(_id)):
+                    _id += 1
+                    with ot_tracer.start_active_span(str(_id)):
+                        pass
+            event.set()
```

Comment:
This line was the culprit. We were running `event.set()` before the context managers closed (which is when the spans are written).

So [occasionally](https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/418/workflows/4fb0f2f2-2699-41e6-8dba-0fe2480f3411/jobs/215297) the second thread would finish executing before the context managers closed and so the spans would be out of order 😄